doc/services/device_mgmt/smp_groups/smp_group_10.rst

```diff
@@ -0,0 +1,330 @@
+    | "err" -> "rc"    | contains the index of the group-based error code. Only appears if       |
+    |                  | non-zero (error condition) when using SMP version 2.                    |
+    +------------------+-------------------------------------------------------------------------+
+    | "rc"             | :c:enum:`mcumgr_err_t` only appears if non-zero (error condition) when  |
+    |                  | using SMP version 1 or for SMP errors when using SMP version 2.         |
+    +------------------+-------------------------------------------------------------------------+
+
+Details on supported groups command
+***********************************
+
+Details on supported groups command allows fetching details on each supported MCUmgr group, such
+as the name and number of handlers. A device can specify an allow list of groups to return details
+on or details on all groups can be returned.
+
+This command is optional, it can be enabled using :kconfig:option:`CONFIG_MCUMGR_GRP_ENUM_DETAILS`.
+The optional name and number of handlers can be enabled/disabled with
+:kconfig:option:`CONFIG_MCUMGR_GRP_ENUM_DETAILS_NAME` and
+:kconfig:option:`CONFIG_MCUMGR_GRP_ENUM_DETAILS_HANDLERS`.
+
+Details on supported groups request
+===================================
+
+Details on supported groups request header fields:
+
+.. table::
+    :align: center
+
+    +--------+--------------+----------------+
+    | ``OP`` | ``Group ID`` | ``Command ID`` |
+    +========+==============+================+
+    | ``0``  | ``10``       | ``2``          |
+    +--------+--------------+----------------+
+
+.. tabs::
+
+   .. group-tab:: Details on all groups
+
+      The command sends an empty CBOR map as data.
+
+   .. group-tab:: Details on specified groups
+
+      CBOR data of request:
+
+      .. code-block:: none
+
+          {
+              (str)"groups" : [
+                  (uint)
+                  ...
+              ]
+          }
+
+      where:
+
+      .. table::
+          :align: center
+
+          +----------+--------------------------------------------------------------+
+          | "groups" | contains a list of the MCUmgr group IDs to fetch details on. |
+          +----------+--------------------------------------------------------------+
+
+Details on supported groups response
+====================================
+
+Details on supported groups response header fields:
+
+.. table::
+    :align: center
+
+    +--------+--------------+----------------+
+    | ``OP`` | ``Group ID`` | ``Command ID`` |
+    +========+==============+================+
+    | ``1``  | ``10``       | ``2``          |
+    +--------+--------------+----------------+
+
+CBOR data of successful response:
+
+.. code-block:: none
+
+    {
+        (str)"groups" : [
+            {
+                (str)"group"          : (uint)
+                (str,opt)"name"       : (str)
+                (str,opt)"handlers"   : (uint)
```

Comment:
I am not sure whether this field makes sense. How could this information be used?

Reply:
Can think of it as a poor version of a version field. E.g. device is released with a custom group that has 3 handlers, then in a future firmware another handler is added, then later on another one is added, so using this you can determine if that handler exists or not without having to call it. Or a firmware might only add in some additional handlers if a Kconfig is set for certain firmware variations